imap_processing/hi/l1a/science_direct_event.py

```diff
@@ -245,86 +252,60 @@ def create_dataset(de_data_list: list, packet_met_time: list) -> xr.Dataset:
+        data_dict["tof_2"].append(event["tof_2"])
+        data_dict["tof_3"].append(event["tof_3"])
         # IMAP-Hi like to keep de_tag value for diagnostic purposes
-        de_tag.append(event["de_tag"])
+        data_dict["de_tag"].append(event["de_tag"])
         # add packet time to ccsds_met list
-        ccsds_met.append(packet_met_time[index])
+        data_dict["ccsds_met"].append(packet_met_time[index])
+
+    # Load the CDF attributes
+    cdf_manager = CdfAttributeManager(imap_module_directory / "cdf" / "config")
+    cdf_manager.load_global_attributes("imap_hi_global_cdf_attrs.yaml")
+    cdf_manager.load_variable_attributes("imap_hi_variable_attrs.yaml")
+
+    # Inject sensor head into Logical_source
+    gattrs = cdf_manager.get_global_attributes("imap_hi_l1a_de_attrs")
+    gattrs["Logical_source"] = gattrs["Logical_source"].format(sensor=sensor_str)
 
     epoch_time = xr.DataArray(
-        de_met_time,
+        data_dict["epoch"],
         name="epoch",
         dims=["epoch"],
         attrs=ConstantCoordinates.EPOCH,
     )
 
     dataset = xr.Dataset(
         coords={"epoch": epoch_time},
-        attrs=hi_cdf_attrs.hi_de_l1a_attrs.output(),
+        attrs=gattrs,
     )
 
-    dataset["esa_step"] = xr.DataArray(
-        np.array(esa_step, dtype=np.uint8),
-        dims="epoch",
-        attrs=hi_cdf_attrs.esa_step_attrs.output(),
-    )
-    dataset["trigger_id"] = xr.DataArray(
-        np.array(trigger_id, dtype=np.uint8),
-        dims="epoch",
-        attrs=hi_cdf_attrs.trigger_id_attrs.output(),
-    )
-    dataset["tof_1"] = xr.DataArray(
-        np.array(tof_1, dtype=np.uint16),
-        dims="epoch",
-        attrs=dataclasses.replace(
-            hi_cdf_attrs.tof_attrs,
-            fieldname="Time of Flight (TOF) 1",
-            label_axis="TOF1",
-        ).output(),
-    )
-    dataset["tof_2"] = xr.DataArray(
-        np.array(tof_2, dtype=np.uint16),
-        dims="epoch",
-        attrs=dataclasses.replace(
-            hi_cdf_attrs.tof_attrs,
-            fieldname="Time of Flight (TOF) 2",
-            label_axis="TOF2",
-        ).output(),
-    )
-    dataset["tof_3"] = xr.DataArray(
-        np.array(tof_3, dtype=np.uint16),
-        dims="epoch",
-        attrs=dataclasses.replace(
-            hi_cdf_attrs.tof_attrs,
-            fieldname="Time of Flight (TOF) 3",
-            label_axis="TOF3",
-        ).output(),
-    )
-    dataset["de_tag"] = xr.DataArray(
-        np.array(de_tag, dtype=np.uint16),
-        dims="epoch",
-        attrs=hi_cdf_attrs.de_tag_attrs.output(),
-    )
-    dataset["ccsds_met"] = xr.DataArray(
-        np.array(ccsds_met, dtype=np.uint32),
-        dims="epoch",
-        attrs=hi_cdf_attrs.ccsds_met_attrs.output(),
-    )
+    for var_name, data in data_dict.items():
+        if var_name == "epoch":
+            continue
+        attrs = cdf_manager.get_variable_attributes(f"hi_de_{var_name}")
+        dtype = attrs.pop("dtype")
```

Comment:
Nice, I like this and the format style above! Seems like a nice clean way to do this.

I assume that `cdf_manager.get_variable_attributes()` returns a copy of the `attrs` dict each time so you aren't mutating/removing the `dtype` from the original version here?

Reply:
Good point... I don't see that a copy is being returned. I will copy here for now.